src/async_api/tokio/io_ops/blocking_io.rs

```diff
@@ -0,0 +1,54 @@
+use super::FileDescriptorRawHandle;
+use async_trait::async_trait;
+use libc::{self, c_int, c_void, size_t};
+use log::error;
+use std::{io, sync::Arc, time::Duration};
+
+#[derive(Debug, Clone)]
+pub struct SubChannel {
+    fd: Arc<FileDescriptorRawHandle>,
+}
+
+#[async_trait::async_trait]
+impl crate::async_api::reply::ReplySender for SubChannel {
+    async fn send(&self, data: &[&[u8]]) {
+        if let Err(err) = SubChannel::send(self, data).await {
+            error!("Failed to send FUSE reply: {}", err);
+        }
+    }
+}
+
+impl SubChannel {
+    pub fn as_raw_fd(&self) -> &FileDescriptorRawHandle {
+        &self.fd
+    }
+
+    pub fn new(fd: FileDescriptorRawHandle, _max_poll_timeout: Duration) -> io::Result<SubChannel> {
+        Ok(SubChannel { fd: Arc::new(fd) })
+    }
+
+    /// Send all data in the slice of slice of bytes in a single write (can block).
+    pub async fn send(&self, buffer: &[&[u8]]) -> io::Result<()> {
+        let iovecs: Vec<_> = buffer
+            .iter()
+            .map(|d| libc::iovec {
+                iov_base: d.as_ptr() as *mut c_void,
+                iov_len: d.len() as size_t,
+            })
+            .collect();
+        let rc = unsafe { libc::writev(self.fd.fd, iovecs.as_ptr(), iovecs.len() as c_int) };
+        if rc < 0 {
+            Err(io::Error::last_os_error())
+        } else {
+            Ok(())
+        }
+    }
+
+    pub fn close(&self) {
```

Comment:
We do call them in drop too if they haven't been done. Calling it upfront ensures the system is torn down/nothing can work once its shutting down. But with the one shot channels in it now... maybe they can be removed and just use drop now. would have to look again

Reply:
(they are all wrapped in RawFileDescriptor handle which does close on drop)

Reply:
Sorry to clarify, I meant why do we need this method? It seems like it could be removed, and the logic moved into `Drop`

Reply:
In macOS we call close explicitly to ensure the blocking libc:read’s on the child sockets terminate. On linux it shouldn’t matter i don’t think. I was keeping the two consistent, but we could move it all to drop for the full async one. (its the blocking read on a socket thats blocking in macOS that i haven’t been able to avoid yet with the macosfuse oddities over fnctl’s)

Reply:
Ah, yes things make sense now. I remember seeing the `cfg` for osx vs linux to select blocking vs non-blocking, but I didn't fully understand the implications. Mind adding a comment about this somewhere? I'll probably have forgotten by the time I have to touch this code next, and don't want to break it :)

Reply:
Added some comment on the place in channel where we call it and the motivation. Lmk for edits/any thoughts though.